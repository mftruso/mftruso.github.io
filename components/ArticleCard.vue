<template>
  <article>
    <v-card class="mx-auto" max-width="400">
      <NuxtLink :to="`projects/${article.slug}`">
        <v-img :src="article.img" :alt="article.img" height="200px" class="align-end" />
        <v-card-title class="text--primary">
          {{ article.title }}
        </v-card-title>
      </NuxtLink>
      <v-card-text>
        <p>{{ article.description }}</p>
      </v-card-text>
      <v-card-actions>
        <v-btn color="primary" :to="`projects/${article.slug}`" text>
          Details
        </v-btn>
        <v-btn v-if="article.link" color="primary" :href="article.link" target="_blank" text>
          View Project <v-icon small>
            mdi-open-in-new
          </v-icon>
        </v-btn>
      </v-card-actions>
    </v-card>
  </article>
</template>

<script>
export default {
  props: { article: Object }
}
</script>
