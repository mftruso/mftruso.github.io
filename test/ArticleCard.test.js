import { describe, test, beforeEach, expect } from 'vitest'
import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import ArticleCard from '@/components/ArticleCard.vue'

describe('ArticleCard', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(ArticleCard, {
      stubs: {
        NuxtLink: RouterLinkStub
      },
      propsData: { article: { title: 'Cool Project', description: 'I learned a lot', _path: 'cool', link: 'https://github.com' } }
    })
  })

  test('Matches Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
    const buttons = wrapper.findAll('v-btn')
    const link = wrapper.findComponent('nuxtlink')
    expect(buttons).toHaveLength(2)
    expect(link).toBeTruthy()
  })
})
