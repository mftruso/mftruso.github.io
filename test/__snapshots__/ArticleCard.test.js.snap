// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard Matches Snapshot 1`] = `
"<article>
  <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" maxwidth=\\"400\\" class=\\"mx-auto\\"><a>
      <v-img-stub height=\\"200px\\" options=\\"[object Object]\\" position=\\"center center\\" src=\\"\\" transition=\\"fade-transition\\" class=\\"align-end\\"></v-img-stub>
      <v-card-title-stub class=\\"text--primary\\">
        Cool Project
      </v-card-title-stub>
    </a>
    <v-card-text-stub>
      <p>I learned a lot</p>
    </v-card-text-stub>
    <v-card-actions-stub>
      <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" to=\\"projects/cool\\" text=\\"true\\" type=\\"button\\">
        Details
      </v-btn-stub>
      <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" href=\\"https://github.com\\" target=\\"_blank\\" text=\\"true\\" type=\\"button\\">
        View Project <v-icon-stub small=\\"\\">
          mdi-open-in-new
        </v-icon-stub>
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</article>"
`;
